nntrainer/layers/add_layer.cpp

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 SeungBaek Hong <sb92.hong@samsung.com>
+ *
+ * @file   add_layer.cpp
+ * @date   2 August 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author SeungBaek Hong <sb92.hong@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is add layer class (operation layer)
+ *
+ */
+
+#include <add_layer.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <util_func.h>
+
+#include <layer_context.h>
+
+namespace nntrainer {
+
+static constexpr size_t SINGLE_INOUT_IDX = 0;
+
+void AddLayer::finalize(InitLayerContext &context) {
+  context.setOutputDimensions({context.getInputDimensions()[0]});
+}
+
+void AddLayer::forwarding(RunLayerContext &context, bool training) {
+  Tensor &hidden_ = context.getOutput(SINGLE_INOUT_IDX);
+
+  const Tensor &input0 = context.getInput(0);
+  const Tensor &input1 = context.getInput(1);
```

Comment:
Does add layer only has 2 inputs?
Is there any possibility of having more than 2 inputs?

Reply:
@lhs8928 Good point! The addition layer, to be precise, implements a summation operation. This layer corresponds to four basic operations that take two inputs.